### PR TITLE
Improve interface for DC Dipole source

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Remove `**kwargs` from the constructor of the class, add `current` as one of the arguments. Improve the logic and error messages for the `location`, `location_a` and `location_b` arguments. Add tests for these new checks. Update docstring of the constructor to reflect latest changes. Fix the expected shape of the `location_a` and `location_b` arguments.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Reference issue

Works towards solving #1391

### CODE REVIEW

File: simpeg/electromagnetics/static/resistivity/sources.py [hunk 1]
```diff
@@ -154,41 +159,46 @@ def __init__(
         location_a=None,
         location_b=None,
         location=None,
-        **kwargs,
+        current=(1.0, -1.0),
```
Comment (jcapriot) [inline simpeg/electromagnetics/static/resistivity/sources.py:162]: I wouldn't support having different currents for each electrode on this class.

Comment (santisoler) [reply simpeg/electromagnetics/static/resistivity/sources.py:162]: That makes total sense! And the sign of the current for each electrode should be always `(+, -)`, right?

Comment (santisoler) [reply simpeg/electromagnetics/static/resistivity/sources.py:162]: If that's so, then I think we should only take a single float as the `current` argument for this method.

File: simpeg/electromagnetics/static/resistivity/sources.py [hunk 2]
```diff
@@ -139,13 +139,17 @@ class Dipole(BaseSrc):
...
-    location_b : (n_source, dim) numpy.array_like
-        B electrode locations; remember to set 'location_a' keyword argument to define M electrode locations.
-    location : list or tuple of length 2 of numpy.array_like
+    location_a : (dim) numpy.array_like
+        A electrode locations; remember to set 'location_b' keyword argument to
+        define N electrode locations.
+    location_b : (dim) numpy.array_like
+        B electrode locations; remember to set 'location_a' keyword argument to
+        define M electrode locations. location : list or tuple of length 2 of
+        numpy.array_like
```
Comment (lheagy) [inline simpeg/electromagnetics/static/resistivity/sources.py:148]:
```suggestion
    location_b : (dim) numpy.array_like
        B electrode locations; remember to set 'location_a' keyword argument to
        define A electrode locations. 
    location : list or tuple of length 2 of
        numpy.array_like
```

File: simpeg/electromagnetics/static/resistivity/sources.py [hunk 3]
```diff
@@ -154,41 +158,46 @@ def __init__(
         location_a=None,
         location_b=None,
         location=None,
-        **kwargs,
```
Comment (lheagy) [inline simpeg/electromagnetics/static/resistivity/sources.py:157]: 🎉

**Review by lheagy — APPROVED** [review]
This looks great Santi! Just one minor comment to address with docstrings, then it is good to go from my perspective 🚀

File: simpeg/electromagnetics/static/resistivity/sources.py (hunk 2 above)
Comment (santisoler) [reply simpeg/electromagnetics/static/resistivity/sources.py:148]: Nice catch. I missed that mistake. Fixing it!

**Pull request merged**